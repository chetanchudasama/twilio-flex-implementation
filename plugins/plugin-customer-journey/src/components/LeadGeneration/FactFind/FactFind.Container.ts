import { connect } from "react-redux";
import { Dispatch } from "redux";

import { FactFind } from "./FactFind";
import { AppState, namespace } from "../../../states";
import { StateToProps, DispatchToProps } from "./FactFind.Props";

const mapStateToProps = (state: AppState): StateToProps => {
  return {
    vehicleSearchDropdownData:
      state[namespace].staticItems.vehicleSearchDropdownData,
    timeForPurchaseItems: state[namespace].staticItems.timeForPurchaseItems,
  };
};

const mapDispatchToProps = (dispatch: Dispatch<any>): DispatchToProps => ({});

export default connect(mapStateToProps, mapDispatchToProps)(FactFind);
