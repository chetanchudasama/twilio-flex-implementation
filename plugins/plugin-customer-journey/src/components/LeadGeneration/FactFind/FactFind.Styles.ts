import styled from "react-emotion";
import { black, darkGrey } from "@common/components";

export const FactFindStyles = styled("div")({
  padding: 24,
  marginLeft: -8,
  width: "100%",
  ".heading": {
    fontSize: 20,
    lineHeight: "24px",
    fontWeight: 600,
    color: black,
    marginBottom: 4,
  },
  ".spacing-left": {
    paddingLeft: 16,
  },
  ".custom-width-select": {
    marginTop: 0,
    marginBottom: 0,
  },
  ".custom-width-select > div": {
    height: 40,
    width: 224,
  },
  ".custom-width-select div div": {
    paddingTop: 12,
    paddingBottom: 12,
  },
  ".custom-input": {
    width: 224,
    marginTop: "0 !important",
    marginBottom: "0 !important",
  },
  ".custom-input > div": {
    marginTop: "0 !important",
    height: 40,
    alignItems: "baseline",
  },
  ".custom-input input": {
    paddingTop: "12px !important",
  },
  ".top-spacing": {
    paddingTop: "0 !important",
  },
  ".optional-text": {
    color: darkGrey,
    fontSize: 12,
  },
});
