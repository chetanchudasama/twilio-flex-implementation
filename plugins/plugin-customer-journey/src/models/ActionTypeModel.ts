export class ActionTypeModel {
  public id: number;

  public name: string;

  constructor() {
    this.id = -1;
    this.name = "";
  }
}
