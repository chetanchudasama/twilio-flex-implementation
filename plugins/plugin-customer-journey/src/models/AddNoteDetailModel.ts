export class AddNoteDetailModel {
  public applicationId: number;

  public content: string;

  public isImportant: boolean;

  public sendToBroker: boolean;

  constructor() {
    this.applicationId = -1;
    this.content = "";
    this.isImportant = false;
    this.sendToBroker = false;
  }
}
