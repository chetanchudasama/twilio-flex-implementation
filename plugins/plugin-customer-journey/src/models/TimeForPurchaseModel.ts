import { ItemModel } from "@common/components";

export class TimeForPurchaseItemModel extends ItemModel {
  constructor() {
    super();
  }
}
