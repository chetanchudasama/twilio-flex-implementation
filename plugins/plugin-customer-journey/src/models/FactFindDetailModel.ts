import { ItemModel } from "@common/components";
import { TimeForPurchaseItemModel } from "./TimeForPurchaseModel";

export class FactFindDetailModel {
  public borrowAmount: number;

  public hasCustomerAlreadyFoundCar: boolean | null;

  public makeId: number;

  public modelId: number;

  public vrm: string;

  public vehicleType: ItemModel;

  public timeForPurchase: TimeForPurchaseItemModel | null;

  public note: string;

  constructor() {
    this.borrowAmount = 0;
    this.hasCustomerAlreadyFoundCar = null;
    this.makeId = -1;
    this.modelId = -1;
    this.vrm = "";
    this.vehicleType = new ItemModel();
    this.timeForPurchase = new TimeForPurchaseItemModel();
    this.note = "";
  }
}
