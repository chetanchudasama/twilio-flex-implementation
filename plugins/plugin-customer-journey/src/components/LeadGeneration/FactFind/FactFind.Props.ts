import { ItemModel } from "@common/components";
import { FactFindDetailModel } from "../../../models/FactFindDetailModel";
import { VehicleSearchDropdownData } from "../../../models/VehicleSearchDropdownData";
import { TimeForPurchaseItemModel } from "../../../models/TimeForPurchaseModel";

export interface StateToProps {
  vehicleSearchDropdownData: VehicleSearchDropdownData;
  timeForPurchaseItems: TimeForPurchaseItemModel[];
}

export interface DispatchToProps {}

export interface OwnProps {
  factFindDetail: FactFindDetailModel;
  updateFactFindDetail: (
    key: string,
    value:
      | string
      | number
      | boolean
      | TimeForPurchaseItemModel
      | ItemModel
      | null
  ) => void;
}

export type FactFindProps = StateToProps & DispatchToProps & OwnProps;
