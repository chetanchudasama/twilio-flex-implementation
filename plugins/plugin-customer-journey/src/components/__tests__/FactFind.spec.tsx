import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";

import { ItemModel } from "@common/components";
import { FactFindDetailModel } from "../../models/FactFindDetailModel";
import { TimeForPurchaseItemModel } from "../../models/TimeForPurchaseModel";
import { FactFindProps } from "../LeadGeneration/FactFind/FactFind.Props";
import { FactFind } from "../LeadGeneration/FactFind/FactFind";

const timeForPurchaseItems: TimeForPurchaseItemModel[] = [
  {
    id: 1,
    name: "timeForPurchaseItem1",
  },
  {
    id: 2,
    name: "timeForPurchaseItem2",
  },
  {
    id: 3,
    name: "timeForPurchaseItem3",
  },
];

const items: ItemModel[] = [
  { id: 1, name: "test1" },
  { id: 2, name: "test2" },
  { id: 3, name: "test3" },
];

const vehicleSearchDropdownData = {
  makesModels: [
    {
      id: 1,
      name: "test1",
      models: [
        { id: 1, name: "test1", count: 1 },
        { id: 2, name: "test2", count: 2 },
      ],
    },
    {
      id: 2,
      name: "test2",
      models: [
        { id: 1, name: "test1", count: 1 },
        { id: 2, name: "test2", count: 2 },
      ],
    },
    {
      id: 3,
      name: "test3",
      models: [
        { id: 1, name: "test1", count: 1 },
        { id: 2, name: "test2", count: 2 },
      ],
    },
  ],
  body: items,
  colour: items,
  fuel: items,
  trans: items,
  distance: [0, 10, 20],
};

const factFindDetailModel: FactFindDetailModel = new FactFindDetailModel();
factFindDetailModel.borrowAmount = 100;
factFindDetailModel.timeForPurchase = {
  id: 1,
  name: "timeForPurchaseItem1",
};
factFindDetailModel.hasCustomerAlreadyFoundCar = true;
factFindDetailModel.makeId = 1;
factFindDetailModel.modelId = 1;
factFindDetailModel.vrm = "";
factFindDetailModel.vehicleType = { id: 1, name: "Car" };
factFindDetailModel.note = "";

const props: FactFindProps = {
  vehicleSearchDropdownData,
  timeForPurchaseItems,
  factFindDetail: factFindDetailModel,
  updateFactFindDetail: jest.fn(),
};

describe("FactFind", () => {
  it("renders correctly on load", () => {
    const wrapper = create(<FactFind {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should call updateFactFindDetail method, when value of borrowAmount text-field is changed", () => {
    const wrapper = mount(<FactFind {...props} />);

    const input = wrapper.find(".borrowAmount").last().find("input").last();
    input.simulate("change", { target: { value: "1000" } });
    wrapper.update();

    expect(props.updateFactFindDetail).toHaveBeenCalled();
  });

  it("should render make, model and vrm fields, when value of hasCustomerAlreadyFoundCar is true", async () => {
    const wrapper = mount(<FactFind {...props} />);
    wrapper.update();

    expect(wrapper.html()).toContain("What is the car make?");
    expect(wrapper.html()).toContain("What is the car model?");
    expect(wrapper.html()).toContain("Does the customer know the VRM?");
  });

  it("should call updateFactFindDetail method, when value of vrm text-field is changed", () => {
    const wrapper = mount(<FactFind {...props} />);

    expect(wrapper.html()).toContain("Does the customer know the VRM?");

    const input = wrapper.find(".vrm").last().find("input").last();
    input.simulate("change", { target: { value: "VRM" } });
    wrapper.update();

    expect(props.updateFactFindDetail).toHaveBeenCalled();
  });

  it("should call updateFactFindDetail method, when value of notes text-field is changed", () => {
    const wrapper = mount(<FactFind {...props} />);

    const input = wrapper.find(".add-note-text-field");
    input.at(0).simulate("change", { target: { value: "Note" } });
    wrapper.update();

    expect(props.updateFactFindDetail).toHaveBeenCalled();
  });
});
