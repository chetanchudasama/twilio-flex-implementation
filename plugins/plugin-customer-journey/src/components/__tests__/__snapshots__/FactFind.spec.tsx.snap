// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactFind renders correctly on load 1`] = `
<div
  className="css-61v6s9"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 heading"
    >
      Fact Find
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
      >
        <div
          className="MuiGrid-item-2"
        >
          <p
            className="MuiTypography-root-98 MuiTypography-caption-108"
          >
            How much is the customer looking to borrow?
          </p>
        </div>
        <div
          className="MuiGrid-item-2 borrowAmount"
        >
          <div
            className="MuiFormControl-root-138 MuiFormControl-marginDense-140 MuiFormControl-fullWidth-141 CustomInputContent-formControl-134 custom-input"
          >
            <label
              className="MuiFormLabel-root-153 MuiFormLabel-filled-157 MuiInputLabel-root-142 MuiInputLabel-formControl-147 MuiInputLabel-animated-150 MuiInputLabel-shrink-149 MuiInputLabel-marginDense-148 CustomInputContent-label-135"
              data-shrink={true}
            >
              
            </label>
            <div
              className="MuiInputBase-root-173 MuiOutlinedInput-root-160 MuiInputBase-formControl-174 MuiInputBase-marginDense-180 MuiInputBase-adornedStart-177 MuiOutlinedInput-adornedStart-163 CustomInputContent-input-136 "
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="MuiPrivateNotchedOutline-root-190 MuiOutlinedInput-notchedOutline-167"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="MuiPrivateNotchedOutline-legend-191"
                  style={
                    Object {
                      "width": 0,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <div
                className="MuiInputAdornment-root-192 MuiInputAdornment-positionStart-194"
              >
                <p
                  className="MuiTypography-root-98 MuiTypography-body1-107 MuiTypography-colorTextSecondary-131"
                >
                  Â£
                </p>
              </div>
              <input
                aria-invalid={false}
                className="MuiInputBase-input-183 MuiOutlinedInput-input-168 MuiInputBase-inputMarginDense-184 MuiOutlinedInput-inputMarginDense-169 MuiInputBase-inputAdornedStart-188 MuiOutlinedInput-inputAdornedStart-171"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value={100}
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <hr
            className="MuiDivider-root-197"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 top-spacing"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
      >
        <div
          className="MuiGrid-item-2"
        >
          <p
            className="MuiTypography-root-98 MuiTypography-caption-108"
          >
            Has the customer found a vehicle yet?
          </p>
        </div>
        <div
          className="MuiGrid-item-2"
        >
          <div
            className="MuiToggleButtonGroup-root-206 ThreeStateToggleContent-root-202 MuiToggleButtonGroup-selected-207"
          >
            <button
              className="MuiButtonBase-root-212 MuiToggleButton-root-208 ThreeStateToggleContent-button-203 btn-intermediate"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
              value={2}
            >
              <span
                className="MuiToggleButton-label-211 ThreeStateToggleContent-label-205"
              >
                Not Asked
              </span>
              <span
                className="MuiTouchRipple-root-240"
              />
            </button>
            <button
              className="MuiButtonBase-root-212 MuiToggleButton-root-208 ThreeStateToggleContent-button-203 btn-no"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
              value={1}
            >
              <span
                className="MuiToggleButton-label-211 ThreeStateToggleContent-label-205"
              >
                No
              </span>
              <span
                className="MuiTouchRipple-root-240"
              />
            </button>
            <button
              className="MuiButtonBase-root-212 MuiToggleButton-root-208 ThreeStateToggleContent-button-203 MuiToggleButton-selected-210 ThreeStateToggleContent-selected-204 btn-yes"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
              value={0}
            >
              <span
                className="MuiToggleButton-label-211 ThreeStateToggleContent-label-205"
              >
                Yes
              </span>
              <span
                className="MuiTouchRipple-root-240"
              />
            </button>
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <hr
            className="MuiDivider-root-197"
          />
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <div
            className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
          >
            <div
              className="MuiGrid-item-2"
            >
              <p
                className="MuiTypography-root-98 MuiTypography-caption-108 spacing-left"
              >
                What is the car make?
              </p>
            </div>
            <div
              className="MuiGrid-item-2"
            >
              <div
                className="MuiFormControl-root-138 MuiFormControl-marginDense-140 MuiFormControl-fullWidth-141 CustomSelectContent-formControl-215 custom-width-select"
              >
                <div
                  className="MuiInputBase-root-173 MuiOutlinedInput-root-160 MuiInputBase-formControl-174 MuiInputBase-marginDense-180 CustomSelectContent-select-217 "
                  onClick={[Function]}
                >
                  <fieldset
                    aria-hidden={true}
                    className="MuiPrivateNotchedOutline-root-190 MuiOutlinedInput-notchedOutline-167"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="MuiPrivateNotchedOutline-legend-191"
                      style={
                        Object {
                          "width": 0,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                  <div
                    className="MuiSelect-root-220"
                  >
                    <div
                      aria-haspopup="true"
                      aria-pressed="false"
                      className="MuiSelect-select-221 MuiSelect-selectMenu-224 MuiInputBase-input-183 MuiOutlinedInput-input-168 MuiInputBase-inputMarginDense-184 MuiOutlinedInput-inputMarginDense-169"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      test1
                    </div>
                    <input
                      type="hidden"
                      value={1}
                    />
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-227 MuiSelect-icon-226"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <hr
                className="MuiDivider-root-197"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <div
            className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
          >
            <div
              className="MuiGrid-item-2"
            >
              <p
                className="MuiTypography-root-98 MuiTypography-caption-108 spacing-left"
              >
                What is the car model?
              </p>
            </div>
            <div
              className="MuiGrid-item-2"
            >
              <div
                className="MuiFormControl-root-138 MuiFormControl-marginDense-140 MuiFormControl-fullWidth-141 CustomSelectContent-formControl-215 custom-width-select"
              >
                <div
                  className="MuiInputBase-root-173 MuiOutlinedInput-root-160 MuiInputBase-formControl-174 MuiInputBase-marginDense-180 CustomSelectContent-select-217 "
                  onClick={[Function]}
                >
                  <fieldset
                    aria-hidden={true}
                    className="MuiPrivateNotchedOutline-root-190 MuiOutlinedInput-notchedOutline-167"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="MuiPrivateNotchedOutline-legend-191"
                      style={
                        Object {
                          "width": 0,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                  <div
                    className="MuiSelect-root-220"
                  >
                    <div
                      aria-haspopup="true"
                      aria-pressed="false"
                      className="MuiSelect-select-221 MuiSelect-selectMenu-224 MuiInputBase-input-183 MuiOutlinedInput-input-168 MuiInputBase-inputMarginDense-184 MuiOutlinedInput-inputMarginDense-169"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      test1
                    </div>
                    <input
                      type="hidden"
                      value={1}
                    />
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-227 MuiSelect-icon-226"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <hr
                className="MuiDivider-root-197"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <div
            className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
          >
            <div
              className="MuiGrid-item-2"
            >
              <p
                className="MuiTypography-root-98 MuiTypography-caption-108 spacing-left"
              >
                Does the customer know the VRM?
                <span
                  className="optional-text"
                >
                  Â (Optional)
                </span>
              </p>
            </div>
            <div
              className="MuiGrid-item-2 vrm"
            >
              <div
                className="MuiFormControl-root-138 MuiFormControl-marginDense-140 MuiFormControl-fullWidth-141 CustomInputContent-formControl-134 custom-input"
              >
                <label
                  className="MuiFormLabel-root-153 MuiInputLabel-root-142 MuiInputLabel-formControl-147 MuiInputLabel-animated-150 MuiInputLabel-shrink-149 MuiInputLabel-marginDense-148 CustomInputContent-label-135"
                  data-shrink={true}
                >
                  
                </label>
                <div
                  className="MuiInputBase-root-173 MuiOutlinedInput-root-160 MuiInputBase-formControl-174 MuiInputBase-marginDense-180 CustomInputContent-input-136 "
                  onClick={[Function]}
                >
                  <fieldset
                    aria-hidden={true}
                    className="MuiPrivateNotchedOutline-root-190 MuiOutlinedInput-notchedOutline-167"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="MuiPrivateNotchedOutline-legend-191"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input-183 MuiOutlinedInput-input-168 MuiInputBase-inputMarginDense-184 MuiOutlinedInput-inputMarginDense-169"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Enter VRM"
                    required={false}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
            >
              <hr
                className="MuiDivider-root-197"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 top-spacing"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
      >
        <div
          className="MuiGrid-item-2"
        >
          <p
            className="MuiTypography-root-98 MuiTypography-caption-108"
          >
            What type of vehicle are they looking for?
          </p>
        </div>
        <div
          className="MuiGrid-item-2"
        >
          <div
            className="MuiFormControl-root-138 MuiFormControl-marginDense-140 MuiFormControl-fullWidth-141 CustomSelectContent-formControl-215 custom-width-select"
          >
            <div
              className="MuiInputBase-root-173 MuiOutlinedInput-root-160 MuiInputBase-formControl-174 MuiInputBase-marginDense-180 CustomSelectContent-select-217 "
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="MuiPrivateNotchedOutline-root-190 MuiOutlinedInput-notchedOutline-167"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="MuiPrivateNotchedOutline-legend-191"
                  style={
                    Object {
                      "width": 0,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <div
                className="MuiSelect-root-220"
              >
                <div
                  aria-haspopup="true"
                  aria-pressed="false"
                  className="MuiSelect-select-221 MuiSelect-selectMenu-224 MuiInputBase-input-183 MuiOutlinedInput-input-168 MuiInputBase-inputMarginDense-184 MuiOutlinedInput-inputMarginDense-169"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Car
                </div>
                <input
                  type="hidden"
                  value={1}
                />
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-227 MuiSelect-icon-226"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <hr
            className="MuiDivider-root-197"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 top-spacing"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
      >
        <div
          className="MuiGrid-item-2"
        >
          <p
            className="MuiTypography-root-98 MuiTypography-caption-108"
          >
            When are they hoping to be driving their new vehicle?
          </p>
        </div>
        <div
          className="MuiGrid-item-2"
        >
          <div
            className="MuiFormControl-root-138 MuiFormControl-marginDense-140 MuiFormControl-fullWidth-141 CustomSelectContent-formControl-215 custom-width-select"
          >
            <div
              className="MuiInputBase-root-173 MuiOutlinedInput-root-160 MuiInputBase-formControl-174 MuiInputBase-marginDense-180 CustomSelectContent-select-217 "
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="MuiPrivateNotchedOutline-root-190 MuiOutlinedInput-notchedOutline-167"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="MuiPrivateNotchedOutline-legend-191"
                  style={
                    Object {
                      "width": 0,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <div
                className="MuiSelect-root-220"
              >
                <div
                  aria-haspopup="true"
                  aria-pressed="false"
                  className="MuiSelect-select-221 MuiSelect-selectMenu-224 MuiInputBase-input-183 MuiOutlinedInput-input-168 MuiInputBase-inputMarginDense-184 MuiOutlinedInput-inputMarginDense-169"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  timeForPurchaseItem1
                </div>
                <input
                  type="hidden"
                  value={1}
                />
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-227 MuiSelect-icon-226"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <hr
            className="MuiDivider-root-197"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <p
        className="MuiTypography-root-98 MuiTypography-caption-108"
      >
        Any additional notes?
        <span
          className="optional-text"
        >
          Â (Optional)
        </span>
      </p>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 notes"
    >
      <div
        className="MuiFormControl-root-138 MuiFormControl-fullWidth-141 add-note-text-field"
      >
        <div
          className="MuiInputBase-root-173 MuiOutlinedInput-root-160 MuiInputBase-fullWidth-182 MuiInputBase-formControl-174 MuiInputBase-multiline-181 MuiOutlinedInput-multiline-166"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-190 MuiOutlinedInput-notchedOutline-167"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-191"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <textarea
            aria-invalid={false}
            className="MuiInputBase-input-183 MuiOutlinedInput-input-168 MuiInputBase-inputMultiline-185 MuiOutlinedInput-inputMultiline-170"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter notes here"
            required={false}
            rows={5}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
